encoding/ssz/query/generalized_index.go

```diff
@@ -0,0 +1,275 @@
+			return 0, fmt.Errorf("container field %q not found: %w", element.Name, err)
+		}
+
+		// Get the chunk count for the current container
+		chunkCount, err := getChunkCount(currentInfo)
+		if err != nil {
+			return 0, fmt.Errorf("chunk count error: %w", err)
+		}
+
+		// Update the generalized index to point to the specified field
+		// root = root * base_index * pow2ceil(chunk_count(container)) + fieldPos
+		root = root*nextPowerOfTwo(chunkCount) + fieldPos
+		currentInfo = fieldSsz
+
+		// Check if a path element is a length field
+		if element.Length {
+			// Length field is only valid for List and Bitlist types
+			if fieldSsz.sszType != List && fieldSsz.sszType != Bitlist {
+				return 0, fmt.Errorf("len() is only supported for List and Bitlist types, got %s", fieldSsz.sszType)
+			}
+			// Length is a uint64 per SSZ spec
+			currentInfo = &SszInfo{sszType: Uint64}
+			// root = root * base_index * pow2ceil(chunk_count(container)) + fieldPos
+			root = root*2 + 1
+			continue
+		}
+
+		if element.Index != nil {
+			switch fieldSsz.sszType {
+			case List:
+				li, err := fieldSsz.ListInfo()
+				if err != nil {
+					return 0, fmt.Errorf("list info error: %w", err)
+				}
+				elem, err := li.Element()
+				if err != nil {
+					return 0, fmt.Errorf("list element error: %w", err)
+				}
+				if *element.Index >= li.Limit() {
+					return 0, fmt.Errorf("index %d out of bounds for list with limit %d", *element.Index, li.Limit())
+				}
+				// Compute chunk position for the element
+				var chunkPos uint64
+				if isBasicType(elem.sszType) {
+					start := *element.Index * itemLengthFromInfo(elem)
+					chunkPos = start / bytesPerChunk
+				} else {
+					chunkPos = *element.Index
+				}
+				innerChunkCount, err := getChunkCount(fieldSsz)
+				if err != nil {
+					return 0, fmt.Errorf("chunk count error: %w", err)
+				}
+				// root = root * base_index * pow2ceil(chunk_count(container)) + fieldPos
+				root = root*listBaseIndex*nextPowerOfTwo(innerChunkCount) + chunkPos
+				currentInfo = elem
+
+			case Vector:
+				vi, err := fieldSsz.VectorInfo()
+				if err != nil {
+					return 0, fmt.Errorf("vector info error: %w", err)
+				}
+				elem, err := vi.Element()
+				if err != nil {
+					return 0, fmt.Errorf("vector element error: %w", err)
+				}
+				if *element.Index >= vi.Length() {
+					return 0, fmt.Errorf("index %d out of bounds for vector with length %d", *element.Index, vi.Length())
+				}
+				var chunkPos uint64
+				if isBasicType(elem.sszType) {
+					start := *element.Index * itemLengthFromInfo(elem)
+					chunkPos = start / bytesPerChunk
+				} else {
+					chunkPos = *element.Index
+				}
+				innerChunkCount, err := getChunkCount(fieldSsz)
+				if err != nil {
+					return 0, fmt.Errorf("chunk count error: %w", err)
+				}
+				// root = root * base_index * pow2ceil(chunk_count(container)) + fieldPos
+				root = root*nextPowerOfTwo(innerChunkCount) + chunkPos
+
+				currentInfo = elem
+
+			case Bitlist:
```

Comment:
```suggestion
			case Bitlist, Bitvector:
```

Can we share the same logic like above?

Reply:
So far, I'll leave with one case per type for clarity. 
The only difference is the base index for the root calculation. We would need to add another condition to disambiguate between Bitvector and Bitlist.

Reply:
Ah yes, base index is different. Then this must be okay 😅